### ISSUE TITLE
Support for SSH ed25519 keys

### ISSUE THREAD

**Issue by bbo2adwuff** [maks/MGit#348]
I am trying to connect to a git server with a password protected ed25519 ssh key.

Seems that password protected keys work since https://github.com/maks/MGit/pull/155.

But I get following error:
```
Clone failed:
ssh://username@server/repo.git:
invalid privatekey: SOMECHARACTERS
```

Is ed25519 supported?

**Comment by bbo2adwuff** [comment]
Too bad. Let's see if I can get another key...
But many thanks for your quick response!

**Comment by bbo2adwuff** [comment]
https://github.com/hierynomus/sshj supports ed25519, while jsch does not support it.

After having a quick look at the code, it seems that it might even be a possibility to switch to sshj.

**Comment by kaeptmblaubaer1000** [comment]
@bbo2adwuff I think a wrapper would have to be written to make JGit compatible.
But otherwise sounds great!

**Comment by bbo2adwuff** [comment]
True, maybe I had a too quick look...

There is also a report for JGit: https://bugs.eclipse.org/bugs/show_bug.cgi?id=520927

**Comment by tutaidalal** [comment]
Is there any update for ed25519  in jSCH? In case i have to switch all the implementation to sshj then its a big effort.

**Comment by maks** [comment]
closed in favour of #545
